test/parallel/xyk-pallet.API.tokenTimeout.test.ts

```diff
@@ -0,0 +1,183 @@
+  const poolEventResult = await getEventResultFromMangataTx(poolCurrencyIssue, [
+    "tokens",
+    "Issued",
+    sudo.keyRingPair.address,
+  ]);
+  const poolAssetId = poolEventResult.data[0].split(",").join("");
+  createdToken = new BN(poolAssetId);
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(createdToken, sudo, defaultCurrencyValue),
+    Assets.mintToken(createdToken, testUser1, defaultCurrencyValue),
+    Assets.mintNative(testUser1),
+    Sudo.sudoAs(
+      sudo,
+      Xyk.createPool(
+        MGA_ASSET_ID,
+        defaultPoolVolumeValue,
+        createdToken,
+        defaultPoolVolumeValue
+      )
+    )
+  );
+});
+
+test("xyk-pallet- GIVEN a token WHEN no token timeout is configured THEN fees are charged as normal", async () => {
+  const api = getApi();
+
+  const timeoutMetadata = await api.query.tokenTimeout.timeoutMetadata();
+
+  if (timeoutMetadata.isEmpty === true) {
+    await testUser1.refreshAmounts(AssetWallet.BEFORE);
+    await sellAsset(
+      testUser1.keyRingPair,
+      MGA_ASSET_ID,
+      createdToken,
+      defaultSwapValue,
+      new BN(1)
+    ).then((result) => {
+      const eventResponse = getEventResultFromMangataTx(result);
+      // eslint-disable-next-line jest/no-conditional-expect
+      expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+    await testUser1.refreshAmounts(AssetWallet.AFTER);
+
+    const tokenFees = testUser1
+      .getAsset(MGA_ASSET_ID)
+      ?.amountBefore.free.sub(
+        testUser1.getAsset(MGA_ASSET_ID)?.amountAfter.free!
+      )
+      .add(defaultSwapValue);
+    // eslint-disable-next-line jest/no-conditional-expect
+    expect(tokenFees).bnGt(new BN(0));
+  }
+});
+
+test("GIVEN a non sudo user WHEN tokenTimeout configuration extrinsic is submitted THEN it fails with RequireSudo", async () => {
+  await updateTimeoutMetadata(testUser1, new BN(20), new BN(200000), [
+    [MGA_ASSET_ID, thresholdValue],
+  ]).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+    expect(eventResponse.data).toContain("RequireSudo");
+  });
+});
+
+test("GIVEN an empty tokenTimeout configuration (all options empty) WHEN sudo submit the extrinsic THEN Tx fails because insuficient params", async () => {
+  const checkEmptyTimeoutConfig = await updateTimeoutMetadata(
+    sudo,
+    new BN(0),
+    new BN(0),
+    null
+  );
+  await waitSudoOperataionFail(
+    checkEmptyTimeoutConfig,
+    "InvalidTimeoutMetadata"
+  );
+});
+
+test("GIVEN a tokenTimeout WHEN periodLength and timeoutAmount are set THEN extrinsic succeed and tokensTimeout is correctly configured", async () => {
+  const setupTimeoutConfig = await updateTimeoutMetadata(
+    sudo,
+    new BN(20),
+    new BN(10000),
+    [
+      [MGA_ASSET_ID, thresholdValue],
+      [createdToken, thresholdValue],
+    ]
+  );
+  await waitSudoOperataionSuccess(setupTimeoutConfig);
```

Comment:
plz, additionally check that the storage value matches with the updated ones.

Reply:
added